express-notes-api/index.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const tempArray = [];
+  for (const id in data.notes) {
+    tempArray.push(data.notes[id]);
+  }
+  res.json(tempArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = parseInt(req.params.id);
+  if (isNaN(id) || id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!data.notes[id]) {
+    res.status(404).json({ error: `cannot find note with id ${id}` });
+  } else {
+    res.json(data.notes[id]);
+  }
+});
+
+app.use(express.json()); // for POST, PUT, PATCH
+
+app.post('/api/notes', (req, res) => {
+  if (Object.keys(req.body).length === 0) {
```

Comment:
Here it would be better to directly check if body has the correct property on it, in this case `content`

Reply:
That makes sense. It looks much better that way.